src/Interpreters/InterpreterSelectWithUnionQuery.cpp

```diff
@@ -31,8 +31,13 @@ namespace ErrorCodes
 }
 
 InterpreterSelectWithUnionQuery::InterpreterSelectWithUnionQuery(
-        const ASTPtr & query_ptr_, ContextPtr context_,
-        const SelectQueryOptions & options_, const Names & required_result_column_names)
+    const ASTPtr & query_ptr_, ContextPtr context_, const SelectQueryOptions & options_, const Names & required_result_column_names)
+    : InterpreterSelectWithUnionQuery(query_ptr_, Context::createCopy(context_), options_, required_result_column_names)
```

Comment:
Perhaps it is more consistent to provide another ctor to IInterpreterUnionOrSelectQuery which does the copying?

Reply:
> provide another ctor to IInterpreterUnionOrSelectQuery which does the copying

It's already done. And yes, we can use that c'tor to be more consistent.

Reply:
Let's see if templating all related c'tors make it better)

Reply:
I guess no. We're allowed to implicitly convert `ContextMutablePtr` to `ContextPtr`. Template will introduce ambiguity.